include/wlr/xwayland.h

```diff
@@ -199,7 +199,8 @@ struct wlr_xwayland_surface {
 		struct wl_signal set_override_redirect;
 		struct wl_signal set_geometry;
 		struct wl_signal ping_timeout;
-	} events;
+		struct wl_signal request_reload_configure;
```

Comment:
I think it would be a lot clearer if we emit a single `set_wm_state` event. This way, compositors can recalculate allowed actions like mutter does.